src/Uno.UI.Runtime.Skia.Wpf/Input/HostPointerHandler.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.Devices.Input;
+using Windows.UI.Core;
+using Uno.Extensions;
+using MouseEventArgs = System.Windows.Input.MouseEventArgs;
+using WpfApplication = System.Windows.Application;
+using WpfWindow = System.Windows.Window;
+using System.Threading;
+using System.Windows;
+using System.Windows.Input;
+using System.Windows.Interop;
+using System.Windows.Media;
+using Windows.UI.Input;
+using MouseDevice = System.Windows.Input.MouseDevice;
+using System.Reflection;
+using Windows.System;
+using Uno.UI.Skia.Platform.Extensions;
+using Uno.Foundation.Logging;
+using UnoApplication = Windows.UI.Xaml.Application;
+using WinUI = Windows.UI.Xaml;
+using WpfCanvas = System.Windows.Controls.Canvas;
+using WpfControl = System.Windows.Controls.Control;
+using WpfFrameworkPropertyMetadata = System.Windows.FrameworkPropertyMetadata;
+using Uno.UI.Runtime.Skia.Wpf.Input;
+using Uno.UI.Runtime.Skia.Wpf.Constants;
+using Uno.UI.Runtime.Skia.Wpf;
+
+namespace Uno.UI.XamlHost.Skia.Wpf
+{
+	internal class HostPointerHandler
+	{
+		private WpfControl _hostControl;
+		private HwndSource _hwndSource;
+		private PointerEventArgs _previous;
+		private readonly IWpfHost _host;
+
+		public HostPointerHandler(IWpfHost host)
+		{
+			if (host is not WpfControl hostControl)
+			{
+				throw new ArgumentException($"{nameof(host)} must be a WPF Control instance", nameof(host));
+			}
+			
+			_hostControl = hostControl;
+
+			_hostControl.MouseEnter += HostOnMouseEnter;
+			_hostControl.MouseLeave += HostOnMouseLeave;
+			_hostControl.MouseMove += HostOnMouseMove;
+			_hostControl.MouseDown += HostOnMouseDown;
+			_hostControl.MouseUp += HostOnMouseUp;
+
+			// Hook for native events
+			_hostControl.Loaded += HookNative;
+
+			void HookNative(object sender, RoutedEventArgs e)
+			{
+				_hostControl.Loaded -= HookNative;
+
+				var win = Window.GetWindow(_hostControl);
+
+				var fromDependencyObject = PresentationSource.FromDependencyObject(win);
+				_hwndSource = fromDependencyObject as HwndSource;
+				_hwndSource?.AddHook(OnWmMessage);
+			}
+			_host = host;
+		}
+
+		public void SetPointerCapture(PointerIdentifier pointer)
+			=> _hostControl.CaptureMouse();
+
+		public void ReleasePointerCapture(PointerIdentifier pointer)
+			=> _hostControl.ReleaseMouseCapture();
+
+		#region Native events
+		private void HostOnMouseEnter(object sender, MouseEventArgs args)
+		{
+			try
+			{
+				_host.XamlRoot?.VisualTree.ContentRoot.InputManager.RaisePointerEntered(_previous = BuildPointerArgs(args));
```

Comment:
is `_host.XamlRoot?.VisualTree.ContentRoot.InputManager` something that can be cached somehow? Pointer events are call very often, every small gain can count :)